e2e-tests/development-runtime/playwright/gatsby-script-off-main-thread.spec.ts

```diff
@@ -1,27 +1,65 @@
 import { test, expect } from "@playwright/test"
 
 const id = {
-  templateLiteral: `inline-script-template-literal-mutation`,
-  dangerouslySet: `inline-script-dangerously-set-mutation`,
+  // Acutual <Script>
+  three: `three`,
+
+  // DOM nodes appended as a result of `<Script>` execution
+  templateLiteral: `mutation-inline-script-template-literal`,
+  dangerouslySet: `mutation-inline-script-dangerously-set`,
+  scriptWithSrc: `mutation-script-with-src`,
 }
 
 test.describe(`off-main-thread scripts`, () => {
-  test(`should load successfully`, async ({ page }) => {
+  test(`should load successfully when the page is visited directly`, async ({
+    page,
+  }) => {
     await page.goto(`/gatsby-script-off-main-thread/`)
 
-    // @ts-ignore
-    const scriptWithSrc = await page.evaluate(() => typeof THREE === `function`)
+    const partytownSnippet = page.locator(`[data-partytown]`)
+    const scriptWithSrc = page.locator(`[id=${id.three}]`)
+    const templateLiteral = page.locator(`[id=${id.templateLiteral}]`)
+    const dangerouslySet = page.locator(`[id=${id.dangerouslySet}]`)
+
+    await expect(partytownSnippet).toContainText(/THREE/) // Forwards configured
+    await expect(templateLiteral).toHaveText(`${id.templateLiteral}: success`) // Template literal inline scripts loaded
+    await expect(dangerouslySet).toHaveText(`${id.dangerouslySet}: success`) // Dangerously set inline scripts loaded
+    await expect(scriptWithSrc).toHaveAttribute(`type`, `text/partytown-x`) // Scripts with sources loaded
+  })
+
+  // This behavior actually works, but I can't figure out how to fix the wait on CSR navigation in Playwright to not be flakey in the CI
+  test.skip(`should load successfully when navigating via Gatsby Link to a page with off-main-thread scripts`, async ({
```

Comment:
We likely need to re-implement something like the `waitForRouteChange` command we have in Cypress.

I put several hours into different approaches but wasn't able to get them to run consistently. Welcome anyone to try if interested (see the commit history for the different things I tried). Keeping the tests in the PR because the test logic itself is fine.

Reply:
Since behavior is blocked by an upstream issue I'll mark this as resolved